### PR TITLE
Change RHEL8 builds from fedora28 to centos8

### LINKED ISSUE #5466: Build on CentOS 8 instead of Fedora 28
We currently build RStudio for Red Hat Enterprise Linux 8, CentOS 8, and Fedora 28 on a Fedora 28 Docker image.

https://github.com/rstudio/rstudio/blob/4d7c019f68ab6a5e05d84e6a73d10b4d829bcbd3/docker/jenkins/Dockerfile.fedora28-x86_64#L1

We did this because we needed RHEL 8 support months before CentOS 8 was ready. However, now that CentOS has been released, we should use it to build RStudio.

https://blog.centos.org/2019/09/centos-8-and-centos-stream-released/

CentOS 8 is much closer to RHEL 8 than Fedora 28 is, and CentOS has a longer support window than Fedora (meaning we can use a CentOS base for a long time). We've also historically built on CentOS and previously built all RedHat binaries on CentOS 5, 6, and 7.

### PR DESCRIPTION
**Pull request by kfeinauer**
Closes #5466

### CODE REVIEW

File: docker/jenkins/Dockerfile.centos8-x86_64 [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
     boost-devel \
     bzip2-devel \
     clang-devel \
-    curl \
     fakeroot \
     fuse-libs \
     gcc \
     gcc-c++ \
     git \
-    gpg \
```
Comment (jmcphers) [inline docker/jenkins/Dockerfile.centos8-x86_64]: We need this to sign builds. I think the corresponding CentOS 8 package is gnupg1.

Comment (kfeinauer) [reply docker/jenkins/Dockerfile.centos8-x86_64]: This is already pulled in implicitly as another package's dependency, but doesn't hurt to explicitly add it! I have done so, with `gnupg2`.

**Comment by jmcphers** [comment]
It specifically needs to be gnupg1, because gnupg2 doesn't support headless signing. 

https://github.com/rstudio/rstudio/blob/78fc6ac01152fa0738461b740e1bf44df816451b/docker/jenkins/sign-release.sh#L47-L51

Does gnupg1 get pulled in as a dependency of another package?

**Comment by kfeinauer** [comment]
Alright @jmcphers, now building and installing `gpg1` from source!

**Pull request merged**